trp_project_scrum/tests/test_trp_project_scrum.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp.tests.common import TransactionCase
+from openerp.exceptions import AccessError
+import random
+import string
+
+
+class TestTrpProjectScrum(TransactionCase):
+
+    def setUp(self):
+        super(TestTrpProjectScrum, self).setUp()
+        # will trigger implied IDS of other groups
+        # let's make believe we are logged in by passing 'login' in data
+        # create a random login just in case we get one that already exists
+        login1 = ''.join(
+            random.choice(string.ascii_uppercase + string.digits)
+            for _ in range(8))
+        password1 = ''.join(random.choice(
+                string.ascii_uppercase + string.digits
+            ) for _ in range(8)
+        )
+        self.collaborator_follower = self.env['res.users'].create({
+            'name': 'collaborator_follower',
+            'groups_id': [(
+                4,
+                self.env.ref('trp_project_scrum.collaborators_followers').id,
+                0
+            )],
+            'login': login1,
+            'password': password1
+        })
+        self.scrumproject_alpha = self.env['project.project'].with_context(
```

Comment:
Also create the projects and tasks as demo data.

Reply:
No, because tests only run when the module has been marked as having demo data anyway.

Reply:
On runbot yes, but what about when a user tries to install the module in a running installation? They should be able to run the tests without the demo data.

Reply:
1. Most standard Odoo tests rely on demo data;
2. As said before, if a module has not been marked as having demo data, tests will not be run anyway (you wpuld have to patch ir_module_module contents to falsely flag the module as having demo data);
3. So basically the existing tests are designed to run on freshly installed databases, with demo data. Because modified demo data could also invalidate the tests. You could argue that that should not happen, that is: if tests fail in a modified database, the database itself has been corrupted. But most tests have not been designed that way, although the idea could be very usefull. You then would need to change the testrunner to always run the tests, even without demo data. Or you could have a global flag on a test class (like we have _post_install) to say it does not need demo data (_need_demo = False ??).

Reply:
While I have implemented all the changes, including this one here is my final argument:

1) I am not sure that is true, I just checked `account` and `sales`. In `account` while there is a demo folder the data in there are not used and in `sales` they create their data in python.

2) I have been running tests without demo data in mostly all the PRs I have made. Can you point me to where the selection of not running the tests if the demo data are not imported? You mean automatically running? Because I use the `--tests-enable` flag.

3) > So basically the existing tests are designed to run on freshly installed databases, with demo data. 

Not sure about this, what if I already have a database running and I just want to install this module? I have proprietary non OCA modules and I want to test whether everything goes well with this new module. If the data are created in demo then I can only test when the database is in demo. If they are created in python runtime then I can test them in any database without any overhead. That's why the `setUp` is there I guess. And probably the pre-install, post install hooks.

4) I still cannot see clear wins from running the tests only in demo mode. In any case I have written the xml and I will fix travis now.